From a3e055e68227a0dc65864e887ba1f7e0ace12ffc Mon Sep 17 00:00:00 2001
From: Angie Wang <angie.wang@windriver.com>
Date: Tue, 1 Sep 2020 00:00:22 -0400
Subject: [PATCH] Allow set public endpoint url for keystone endpoints

[ Use public endpoint for admin url]
Signed-off-by: Lucas de Ataides <lucas.deataidesbarreto@windriver.com>
[ upversioned of changes to openstack-helm for epoxy release ]
Signed-off-by: cfigueir <carlos.figueiredofelipe@windriver.com>
---
 keystone/templates/job-db-sync.yaml | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/keystone/templates/job-db-sync.yaml b/keystone/templates/job-db-sync.yaml
index a4ff67d80..37db4464b 100644
--- a/keystone/templates/job-db-sync.yaml
+++ b/keystone/templates/job-db-sync.yaml
@@ -23,9 +23,17 @@ helm.sh/hook-weight: "-4"
 {{- $envAll := index . 0 -}}
 env:
   - name: OS_BOOTSTRAP_ADMIN_URL
+    {{- if and (hasKey $envAll.Values.endpoints.identity "force_public_endpoint") $envAll.Values.endpoints.identity.force_public_endpoint }}
+    value: {{ tuple "identity" "public" "api" $envAll | include "helm-toolkit.endpoints.keystone_endpoint_uri_lookup" }}
+    {{- else }}
     value: {{ tuple "identity" "admin" "api" $envAll | include "helm-toolkit.endpoints.keystone_endpoint_uri_lookup" }}
+    {{- end }}
   - name: OS_BOOTSTRAP_INTERNAL_URL
+    {{- if and (hasKey $envAll.Values.endpoints.identity "force_public_endpoint") $envAll.Values.endpoints.identity.force_public_endpoint }}
+    value: {{ tuple "identity" "public" "api" $envAll | include "helm-toolkit.endpoints.keystone_endpoint_uri_lookup" }}
+    {{- else }}
     value: {{ tuple "identity" "internal" "api" $envAll | include "helm-toolkit.endpoints.keystone_endpoint_uri_lookup" }}
+    {{- end }}
   - name: OS_BOOTSTRAP_PUBLIC_URL
     value: {{ tuple "identity" "public" "api" $envAll | include "helm-toolkit.endpoints.keystone_endpoint_uri_lookup" }}
   - name: OPENSTACK_CONFIG_FILE
-- 
2.43.0

